src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1173,20 +1179,19 @@ private ISwitchStatement CreateBoundPatternSwitchStatementOperation(BoundPattern
 
         private ICaseClause CreateBoundPatternSwitchLabelOperation(BoundPatternSwitchLabel boundPatternSwitchLabel)
         {
-            LabelSymbol label = boundPatternSwitchLabel.Label;
             bool isInvalid = boundPatternSwitchLabel.HasErrors;
             SyntaxNode syntax = boundPatternSwitchLabel.Syntax;
             ITypeSymbol type = null;
             Optional<object> constantValue = default(Optional<object>);
 
             if (boundPatternSwitchLabel.Pattern.Kind == BoundKind.WildcardPattern)
             {
-                // Default switch label in pattern switch statement is represented as a regular case clause.
-                Lazy<IOperation> value = new Lazy<IOperation>(() => null);
-                return new LazySingleValueCaseClause(value, BinaryOperationKind.None, CaseKind.Default, isInvalid, syntax, type, constantValue);
+                // Default switch label in pattern switch statement is represented as a default case clause.
+                return new DefaultCaseClause( isInvalid, syntax, type, constantValue);
```

Comment:
I didn't find a test for this case (discard pattern being represented as a DefaultCaseClause).